crud/get.lua

```diff
@@ -14,8 +14,65 @@ local get = {}
 
 local GET_FUNC_NAME = '_crud.get_on_storage'
 
-local function get_on_storage(space_name, key)
-    dev_checks('string', '?')
+local function get_field_metadata(full_metadata, field)
+    dev_checks('table', 'string')
+
+    for _, tuple in ipairs(full_metadata) do
+        if tuple['name'] == field then
+            return tuple
+        end
+    end
+end
+
+local function format_metadata(full_metadata, fields)
+    dev_checks('table', '?table')
+
+    local metadata = {}
+
+    for i, field in ipairs(fields) do
+        metadata[i] = get_field_metadata(full_metadata, field)
+    end
+
+    return metadata
+end
+
+local function format_result_by_fields(formatted_result, fields)
+    dev_checks('table', '?table')
+
+    local result = {}
+
+    result.rows = formatted_result.rows
+    if fields ~= nil then
+        result.metadata = format_metadata(formatted_result.metadata, fields)
+    else
+        result.metadata = formatted_result.metadata
+    end
+
+    return result
+end
+
+local function get_partial_result(func_get_res, fields)
+    dev_checks('table', '?table')
+
+    local result = {}
+
+    result.err = func_get_res.err
+    if func_get_res.res ~= nil then
+        if fields ~= nil then
+            result.res = {}
+            for i, field in ipairs(fields) do
+                result.res[i] = func_get_res.res[field]
+            end
+        else
+            result.res = func_get_res.res
+        end
+    end
```

Comment:
It confuses me a bit. I suggest to pass "tuple" and "fields" as arguments.

If fields are nil then function returns tuple immediately. Otherwise you can extract fields.

I'd recommend to untie your feature as max as possible from schema manipulations and retries logic.